src/components/Calculator.js

```diff
@@ -1,186 +1,86 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Calculator.css';
 import calculate from '../logic/calculator';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-    this.handleClick = this.handleClick.bind(this);
-  }
+const Calculator = () => {
+  const [state, setState] = useState({});
 
-  handleClick = (buttonName) => {
-    this.setState((state) => calculate(state, buttonName));
+  const handleClick = (e) => {
+    setState({ ...state, ...calculate(state, e.target.name) });
   };
 
-  displayResult = () => {
-    const { total, next, operation } = this.state;
-    const display = (total || '') + (operation || '') + (next || '');
+  const { total, next, operation } = state;
+  const display = (total || '') + (operation || '') + (next || '');
 
-    return display || '0';
-  };
```

Comment:
- [ ] OPTIONAL: The app is not working as expected. It has bugs and errors. I don't know if it happens after the refactor but I highly recommend you fix these issues otherwise your repo will create a wrong impression about your skills. Kindly make changes even though it is not required for your project submission. 
- Bug 1: When I click a number and then the `=` operator I get 
![image](https://user-images.githubusercontent.com/64894927/129353958-dceee640-e819-4b1b-ba8f-f6782fc4bfd9.png)
- Error: When I click the `=` operator twice I get
![Screenshot from 2021-08-13 17-56-10](https://user-images.githubusercontent.com/64894927/129354043-b1a61479-f09d-4a75-9068-386815e1e279.png)
- Bug: When I click a number and the `-` operator and the `+-` operator I get 
![image](https://user-images.githubusercontent.com/64894927/129354156-eeb80fe9-ff1a-471a-a811-e41a64e20915.png)
- Bug: The `%` operator does not work properly

Reply:
Thank you, I did not check this. I will fix it 😄